daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -491,36 +491,41 @@ private[lf] final class Compiler(
         env.toSEVar(cidPos),
         SBFetchAny(env.toSEVar(cidPos), s.SEValue.None),
       ),
-    ) { (payloadPos, _env) =>
-      val env = _env.bindExprVar(param, payloadPos).bindExprVar(choice.argBinder._1, choiceArgPos)
+    ) { (payloadPos, env) =>
       let(
         env,
-        SBApplyChoiceGuard(choice.name, Some(ifaceId))(
-          env.toSEVar(guardPos),
-          env.toSEVar(payloadPos),
-          env.toSEVar(cidPos),
-        ),
-      ) { (_, env) =>
+        s.SEPreventCatch(SBViewInterface(ifaceId)(env.toSEVar(payloadPos))),
+      ) { (_, _env) =>
+        val env = _env.bindExprVar(param, payloadPos).bindExprVar(choice.argBinder._1, choiceArgPos)
```

Comment:
Why is `env` extended by hand here (and in some other places)
but not in `compileFetchInterface` following a similar call to `s.SEPreventCatch(SBViewInterface ...`
?

Reply:
Because the variable `param` (this in surface language) and `choice.argBinder._1` (choice argument) should not be in the scope for computing controller, observer, and view.

There is no variable involve in the case of fetch.

Reply:
Thank you for clarifying this!